### DISCUSSION TITLE
How should file tangling details be specified?

### DISCUSSION THREAD

**Discussion by Luis-Henriquez-Perez** [aldrichtr/tangld#19] [General]
Right now I'm thinking of an list of lists.

`(regexp-or-fn list-of-paths-to-tangle-to whether-to-tangle-or-not)`

There should also be a variable for the directory to link to (most of the time it will be home I'm guessing)

Once the files in the dotfiles directory should then be linked to the home directory (or whatever the user specifies).

There should be an option to write this information to a file somehow so it can be version controlled and read.

**Comment by Luis-Henriquez-Perez** [comment]
I decided just to have a user option that takes a function. The function will take a file path and decide what to do with it. It could do nothing, or it could tangle it to the dotfiles dir, or it could just link it to the dotfiles dir. Whatever the user wants.

**Comment by aldrichtr** [comment]
I think what you are saying is that you want to specify the :target <filename> portion of the header args for a src block from a function in tangld?  I like that idea.  Originally, I was thinking that within the org file, the user would specify the target like so:

> #+PROPERTY: header-args+ :tangle yes
> #+PROPERTY: header-args+ :tangle ~/.tmux.conf
these values can also have a relative directory, or a path below a root-directory if we use `ob-extended-tangle`.  However, it may also be useful to be able to "redirect" that target to a test or staging area, is that what you were thinking?

**Comment by Luis-Henriquez-Perez** [comment]
> I like that idea. Originally, I was thinking that within the org file

Ok. Yea good point.

Also, we need to consider non-org files. It is not unlikely that the user will have at least a few non-org files which (because they're not org files) cannot be tangled. So we need to give the user a way to account for those.

**Comment by Luis-Henriquez-Perez** [comment]
> these values can also have a relative directory, or a path below a root-directory if we use ob-extended-tangle. However, it may also be useful to be able to "redirect" that target to a test or staging area, is that what you were thinking?

Yes, I have been thinking about this more. I'd like users to have the ability to "redirect" these files wherever they want. They'd be able to specify a function which takes a target such as ~/.tmux.conf and return the path it should be tangled to.

**Comment by aldrichtr** [comment]
I like that idea. So we need to parse an org file, gather up the tangle options embedded in it and then do some manipulation?
```
defun org-babel--get-vars (params)
  "Return the babel variable assignments in PARAMS.

PARAMS is a quasi-alist of header args, which may contain
multiple entries for the key `:var'.  This function returns a
list of the cdr of all the `:var' entries."
```

Looks like we may be able to use this function out of ob-core ?
